internal/functions/deploy/deploy.go

```diff
@@ -89,6 +97,86 @@ func GetFunctionSlugs(fsys afero.Fs) (slugs []string, err error) {
 	return slugs, nil
 }
 
+// getRemoteFunctions gets the list of functions from the Supabase project
+func getRemoteFunctions(ctx context.Context, projectRef string) ([]string, error) {
+	resp, err := utils.GetSupabase().V1ListAllFunctionsWithResponse(ctx, projectRef)
+	if err != nil {
+		return nil, errors.Errorf("failed to list remote functions: %w", err)
+	}
+
+	if resp.JSON200 == nil {
+		return nil, errors.New("Unexpected error retrieving functions: " + string(resp.Body))
+	}
+
+	var remoteSlugs []string
+	for _, function := range *resp.JSON200 {
+		remoteSlugs = append(remoteSlugs, function.Slug)
+	}
+
+	return remoteSlugs, nil
+}
+
+// pruneFunctions deletes functions that exist remotely but not locally
+func pruneFunctions(ctx context.Context, localSlugs []string, projectRef string, force bool) error {
```

Comment:
I will move this pkg/function so it can extend EdgeRuntimeAPI with a batch delete operation.

Reply:
not doing this anymore due to inherent race condition in api